lib/iris/aux_factory.py

```diff
@@ -76,6 +76,107 @@ def dependencies(self):
 
         """
 
+    @abstractmethod
+    def _calculate_array(self, *dep_arrays, **other_args):
+        """Make a coordinate array from a complete set of dependency arrays.
+
+        Parameters
+        ----------
+        * dep_arrays : tuple of array-like
+            Arrays of data for each dependency.
+            Must match the number of declared dependencies.  All pre-aligned to the
```

Comment:
NOTE - 
have now relaxed this.  Only require them to be broadcastable...
TODO: fix the docstring